<launch>
    <!-- Launch file which enables the estimation filter -->

    <arg name="imu_node" default="imu_3dm_gx4"/>
    <arg name="baudrate" default="921600" />
    <arg name="device" default="/dev/ttyACM0" />

    <!-- Run both at 100Hz by default -->
    <arg name="imu_decimation" default="10"/>
    <arg name="filter_decimation" default="5"/>

    <node pkg="imu_3dm_gx4" name="$(arg imu_node)" type="imu_3dm_gx4" output="screen">
        <param name="device" type="string" value="$(arg device)" />
        <param name="baudrate" type="int" value="$(arg baudrate)" />

        <param name="imu_decimation" type="int" value="$(arg imu_decimation)" />
        <param name="filter_decimation" type="int" value="$(arg filter_decimation)" />

        <param name="enable_filter" type="bool" value="true"/>
        <param name="enable_mag_update" type="bool" value="true"/>
    </node>
</launch>
